<launch>
    <param name="use_sim_time" value="false"/>

    <node pkg="rviz" type="rviz" name="rviz" output="screen" args="-d $(find driving_simulator)/src/simulator.rviz " />
    <node name="map_server" pkg="map_server" type="map_server" args="$(find driving_simulator)/src/map.yaml" output="screen"/>
    <node name="tf" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 1 /map /my_frame 10"/>
    <arg name="model_driver_config" default="$(find driving_simulator)"/>

    <!--Publish the Marker pose -->
    <node name="DESPOT" pkg="driving_simulator" type="DESPOT" respawn="false" output="log"/>
    <node name="DESPOT1" pkg="driving_simulator" type="DESPOT1" respawn="false" output="log"/>
    <node name="DESPOT2" pkg="driving_simulator" type="DESPOT2" respawn="false" output="log"/>
    <node name="DESPOT3" pkg="driving_simulator" type="DESPOT3" respawn="false" output="log"/>
    <node name="MultipolicyMPC" pkg="driving_simulator" type="MultipolicyMPC" respawn="false" output="log"/>

    <group ns="Driver1">
        <node name="DriverModel" pkg="driving_simulator" type="DriverModel_node" respawn="false" output="screen"/>
        <rosparam command="load" file="$(arg model_driver_config)/config/driver1.yaml" />
        <remap from="Driver1/state_A" to="/state_A"/>
        <remap from="Driver1/action" to="/action"/>
    </group>

    <group ns="obs_vehicle_1">
        <node name="DriverModel" pkg="driving_simulator" type="DriverModel_node" respawn="false" output="screen"/>
        <rosparam command="load" file="$(arg model_driver_config)/config/driver2.yaml" />
        <remap from="state_A" to="/state_A_1"/>
        <remap from="action" to="/action_1"/>
    </group>

    <group ns="obs_vehicle_2">
        <node name="DriverModel" pkg="driving_simulator" type="DriverModel_node" respawn="false" output="screen"/>
        <rosparam command="load" file="$(arg model_driver_config)/config/driver3.yaml" />
        <remap from="state_A" to="/state_A_2"/>
        <remap from="action" to="/action_2"/>
        </group>

    <group ns="obs_vehicle_3">
        <node name="DriverModel" pkg="driving_simulator" type="DriverModel_node" respawn="false" output="screen"/>
        <rosparam command="load" file="$(arg model_driver_config)/config/driver4.yaml" />
        <remap from="state_A" to="/state_A_3"/>
        <remap from="action" to="/action_3"/>
    </group>

</launch>
